openpype/hosts/maya/plugins/publish/collect_arnold_scene_source.py

```diff
@@ -18,18 +18,24 @@ def process(self, instance):
         for objset in objsets:
             objset = str(objset)
             members = cmds.sets(objset, query=True)
+            members = cmds.ls(members, long=True)
             if members is None:
                 self.log.warning("Skipped empty instance: \"%s\" " % objset)
                 continue
             if objset.endswith("content_SET"):
-                members = cmds.ls(members, long=True)
                 children = get_all_children(members)
-                instance.data["contentMembers"] = children
-                self.log.debug("content members: {}".format(children))
+                instance.data["contentMembers"] = children + members
+                self.log.debug(
+                    "content members: {}".format(
+                        instance.data["contentMembers"]
+                    )
+                )
             elif objset.endswith("proxy_SET"):
-                set_members = get_all_children(cmds.ls(members, long=True))
-                instance.data["proxy"] = set_members
-                self.log.debug("proxy members: {}".format(set_members))
+                children = get_all_children(cmds.ls(members, long=True))
```

Comment:
Members already consists of long paths.
```suggestion
                children = get_all_children(members)
```

Reply:
Also be aware that the if the `members` also contains one of its own children then `children` will also contain that node. As such `members + children` can result in duplicate entries for the same node. It might be useful to still ensure they are unique afterwards if that's a case we might hit.

Additionally, the logging of the members (which was already there) can now be a much much larger list since it also includes ALL children which can make the logging heavy and slow. We might want to avoid the logging by default **or** like before only log the direct set members instead?

Reply:
not sure if it is quite cliche to do but maybe we can have a function to store the data and use it in the main function.
```
def get_ass_data(objset, suffix):
    if objset.endswith(suffix):
        members = cmds.ls(members, long=True)
        if not members:
            return
        children = get_all_children(members)
        return children

```